src/cpp/polars/TimeSeries.h

```diff
@@ -0,0 +1,143 @@
+//
+// Created by Linda Uruchurtu on 03/10/2018.
+//
+
+#ifndef POLARS_TIMESERIES_H
+#define POLARS_TIMESERIES_H
+
+#include <cassert>
+#include <vector>
+#include <cmath>
+#include <string>
+#include "armadillo"
+#include <chrono>
+#include "date/date.h"
+
+#include "Series.h"
+
+
+namespace polars {
+
+    using namespace std::chrono;
+    typedef std::chrono::duration<double> unix_epoch_seconds;
+
+    template<class TimePointType>
+    class TimeSeries : public Series {
+
+    public:
+        using Series::loc;
+
+        TimeSeries() = default;
+
+        TimeSeries(arma::vec v0, std::vector<TimePointType> t0){
+            v = v0;
+            t = chrono_to_double_vector(t0);
```

Comment:
use the initializer list ` : v(v0), t(chrono_to_double(t0))` as it saves un unnecessary copy (and is more c++-onic)

Reply:
That fails as you cannot initialize base class members from initializer list directly.

Reply:
Ah - you need to call the base class constructor then. `TimeSeries(arma::vec v0, std::vector<TimePointType> t0) : Series(v0, chrono_to_double_vector(t0)) {}``